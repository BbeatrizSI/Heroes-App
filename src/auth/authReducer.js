import { types } from "../types/types";


// const state = {
//     name: 'BÃ¡rbara',
//     logged: true
// }

export const authReducer = (state = {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
    
        case types.logout:
            return {
                logged: false
            }
            
        default:
            return state;
    }
};

