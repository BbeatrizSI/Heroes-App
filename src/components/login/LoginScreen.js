import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

const LoginScreen = ({ history }) => {

    const {dispatch} = useContext(AuthContext);

    const handleClick = () => {

        const lastPath = localStorage.getItem('lastPath') || '/';
            
        dispatch({
            type: types.login,
            payload: {
                name: 'BÃ¡rbara'
            }
        }) 
        
        history.replace(lastPath);
    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>
            <button className="btn btn-primary" onClick= {handleClick}>
Login
            </button>
        </div>
    );
};

export default LoginScreen;