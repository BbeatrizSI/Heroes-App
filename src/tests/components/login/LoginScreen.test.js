import React from 'react';
import { mount } from 'enzyme';
import  LoginScreen  from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';



describe('Test in <LoginScreen />', () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Bárbara'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value ={ contextValue }>
            <LoginScreen history={ history }/>
        </AuthContext.Provider>
    );

   
    test('should show correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
               
    });

    test('should show correctly component', () => {
        
       const handleClick = wrapper.find('button').prop('onClick');

       handleClick();

       expect( contextValue.dispatch ).toHaveBeenCalledWith({
           type: types.login,
           payload: {
               name: 'Bárbara'
           }
       })

       expect ( history.replace ).toHaveBeenCalledWith('/');
       localStorage.setItem('lastPath','/dc');
       handleClick();
       expect ( history.replace ).toHaveBeenCalledWith('/dc');
               
    });

    
})