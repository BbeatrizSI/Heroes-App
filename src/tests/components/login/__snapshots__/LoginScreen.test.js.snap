// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen /> should show correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
